### PR TITLE
docs: documentation on the new scheduler of TiCDC

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 

Issue Number: ref #722

- The need of documentation on the interaction between the owner & processor.

### What is changed and how it works?
- Added documentation.

### Check List 

Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: docs/distributed-scheduling.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# Distributed Scheduling in TiCDC
+
+## Background
+
+TiCDC boasts high availability and horizontal scalability. To make this possible, TiCDC needs a distributed scheduling mechanism, a mechanism by which tables can be distributed across the nodes in the cluster and sustain failures of nodes. We still need a center of control, which we call the Owner, but the Owner should fail over very quickly should the old one fails.
```
Comment (asddongmen) [inline docs/distributed-scheduling.md:5]: "but the Owner should fail over very quickly `should` the old one fails. "
Should we use `when` here?

Comment (liuzix) [inline docs/distributed-scheduling.md:5]:
```suggestion
TiCDC boasts high availability and horizontal scalability. To make this possible, TiCDC needs a distributed scheduling mechanism, a mechanism by which tables can be distributed across the nodes in the cluster and sustain failures of nodes. We still need a center of control, which we call the Owner, but the Owner should fail over very quickly should the old one fail.
```

Comment (liuzix) [reply docs/distributed-scheduling.md]: Please refer to https://www.oxfordlearnersdictionaries.com/definition/english/should?q=should meaning 14

**Review by asddongmen — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**